packages/application-shell/src/components/authenticated/authenticated.js

```diff
@@ -1,17 +1,54 @@
+import React from 'react';
 import PropTypes from 'prop-types';
+import { Query } from 'react-apollo';
 import * as storage from '@commercetools-frontend/storage';
+import { GRAPHQL_TARGETS } from '@commercetools-frontend/constants';
 import { STORAGE_KEYS } from '../../constants';
+import AmILoggedInQuery from './authenticated.graphql';
 
-const getIsAuthenticated = () =>
+const hasCachedAuthenticationState = () =>
   storage.get(STORAGE_KEYS.IS_AUTHENTICATED) === 'true';
 
 const Authenticated = props => {
-  const isAuthenticated = getIsAuthenticated();
-  return props.children({ isAuthenticated });
+  // We attempt to see if the user was already authenticated by looking
+  // at the "cached" flag in local storage.
+  const cachedAuthenticationState = hasCachedAuthenticationState();
+  if (cachedAuthenticationState) {
+    return props.render({ isAuthenticated: true });
+  }
```

Comment:
Still any subsequent request will fail in the mc-be in turn logging the user out again. So there is no harm in this caching here, correct?

Reply:
Yes correct